vulnerabilities/importers/openssl.py

```diff
@@ -20,101 +20,167 @@
+    """
+
+    purl = PackageURL(type="openssl", name="openssl")
+    cve = advisory_url = severity = summary = None
+    safe_pkg_versions = {}
+    vuln_pkg_versions_by_base_version = {}
+    aliases = []
+    references = []
+    affected_packages = []
+    date_published = xml_issue.attrib["public"].strip()
+
+    for info in xml_issue:
+        if info.tag == "impact":
+            severity = VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["generic_textual"], value=info.attrib["severity"]
+            )
+
+        elif info.tag == "advisory":
+            advisory_url = info.attrib["url"]
+            if not advisory_url.startswith("https://web.archive.org"):
+                advisory_url = urljoin("https://www.openssl.org", advisory_url)
+
+        elif info.tag == "cve":
+            cve = info.attrib.get("name")
+            # use made up alias to compensate for case when advisory doesn't have CVE-ID
+            madeup_alias = f"VC-OPENSSL-{date_published}"
+            if cve:
+                cve = f"CVE-{cve}"
+                aliases.append(f"{madeup_alias}-{cve}")
+                aliases.append(cve)
+            else:
+                aliases.append(madeup_alias)
+
+        elif info.tag == "affects":
+            affected_base = info.attrib["base"]
+            affected_version = info.attrib["version"]
+            if affected_base.startswith("fips"):
+                logger.error(f"Unsupported OpenSSL fips. Use a different importer.")
+                return
+            if affected_base in vuln_pkg_versions_by_base_version:
+                vuln_pkg_versions_by_base_version[affected_base].append(affected_version)
+            else:
+                vuln_pkg_versions_by_base_version[affected_base] = [affected_version]
+
+        elif info.tag == "fixed":
+            fixed_base = info.attrib["base"]
+            fixed_version = info.attrib["version"]
+            safe_pkg_versions[fixed_base] = fixed_version
+            for commit in info:
+                commit_hash = commit.attrib["hash"]
+                references.append(
+                    Reference(
+                        url=urljoin("https://github.com/openssl/openssl/commit/", commit_hash)
+                    )
                 )
-                advisories.append(advisory)
 
-        return advisories
+        elif info.tag == "description":
+            summary = " ".join(info.text.split())
+
+        elif info.tag in ("reported", "problemtype", "title"):
+            # as of now, these info isn't useful for AdvisoryData
```

Comment:
Please create a follow up issue to revisit this in the future and paste the link there as a comment

Reply:
https://github.com/nexB/vulnerablecode/issues/688